lib/FieldLocation/FieldLocationFinalContainer.js

```diff
@@ -0,0 +1,82 @@
+import React, {
+  useState,
+  useCallback,
+  useEffect,
+} from 'react';
+import PropTypes from 'prop-types';
+import {
+  find,
+  uniqBy,
+} from 'lodash';
+
+import {
+  Loading,
+} from '@folio/stripes/components';
+
+import {
+  getLocationOptions,
+} from '../utils';
+import FieldLocationFinal from './FieldLocationFinal';
+
+const FieldLocationFinalContainer = ({
+  isDisabled,
+  labelId,
+  locationLookupLabel,
+  locationsForDict,
+  name,
+  onChange,
+  prepopulatedLocationsIds,
+  required,
+}) => {
+  const [locationsForSelect, setLocationsForSelect] = useState();
+
+  useEffect(() => {
+    const locations = prepopulatedLocationsIds?.map(id => find(locationsForDict, { id })) || [];
+
+    setLocationsForSelect(getLocationOptions(locations));
+  },
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
the same

Reply:
In this case `useEffect` plays role `componentDidMount`